front/src/modules/simulationResult/components/SpaceTimeChart/WaypointsModal.tsx

```diff
@@ -0,0 +1,171 @@
+  const openModal = () => {
+    modalRef.current?.showModal();
+  };
+
+  const closeModal = () => {
+    modalRef.current?.close();
+    setWaypointsModalOpen(false);
+  };
+
+  useModalFocusTrap(modalRef, closeModal);
+
+  // TODO (#8628): handle saving the manchette in local storage to keep the current selected waypoints
+  const handleSubmit = () => {
+    if (selectedWaypoints.size < 2) {
+      setIsNotEnoughSelectedWaypoints(true);
+      return;
+    }
+    setFilteredWaypoints(waypoints.filter((_, i) => selectedWaypoints.has(i)));
+    closeModal();
+  };
+
+  useEffect(() => {
+    if (waypointsModalOpen) {
+      openModal();
+    }
+  }, [waypointsModalOpen]);
+
+  // useEffect(() => {
+  //   setSelectedWaypoints(new Set(filteredWaypoints.map((_, i) => i)));
+  // }, [filteredWaypoints]);
+
+  const handleCancel = () => {
+    setSelectedWaypoints(new Set(filteredWaypoints.map((_, i) => i)));
+    closeModal();
+  };
+
+  // TODO : fix checkbox component in osrd-ui to handle shift + click because of a firefox bug (#510)
+  const handleWaypointClick = (event: React.MouseEvent<HTMLInputElement>, index: number) => {
+    event.stopPropagation();
+    if (event.shiftKey && lastSelectedIndex !== null) {
+      const minIndex = Math.min(index, lastSelectedIndex);
+      const maxIndex = Math.max(index, lastSelectedIndex);
+      const newSet = new Set(selectedWaypoints);
+      for (let i = minIndex; i <= maxIndex; i += 1) {
+        newSet.add(i);
+      }
+      setSelectedWaypoints(newSet);
+    } else {
+      setSelectedWaypoints((prev) => {
+        const newSet = new Set(prev);
+        if (newSet.has(index)) {
+          newSet.delete(index);
+        } else {
+          newSet.add(index);
+        }
+        return newSet;
+      });
+    }
+    setLastSelectedIndex(index);
+  };
+
+  const handleMultiCheckboxClick = () => {
+    if (!allWaypointsSelected) {
+      setSelectedWaypoints(new Set(waypoints.map((_, i) => i)));
+    }
+    if (allWaypointsSelected) {
+      setSelectedWaypoints(new Set());
+    }
+  };
+
+  return (
+    <dialog ref={modalRef} className={cx('waypoints-modal')}>
+      <div className="waypoints-modal-header">
+        <div className="name">{t('simulation:waypointsModal.name')}</div>
+        <div className="ch">{t('simulation:waypointsModal.ch')}</div>
+      </div>
+      <div className="waypoints-modal-body">
+        <div className="waypoint-item selector-all">
+          <Checkbox
+            small
+            isIndeterminate={isIndeterminate}
+            checked={allWaypointsSelected}
+            onChange={handleMultiCheckboxClick}
+          />
+        </div>
+        {waypoints.map((waypoint, index) => (
+          <div className="waypoint-item" key={`${waypoint.id}-${index}`}>
+            <Checkbox
+              small
+              checked={selectedWaypoints.has(index)}
+              onChange={() => {}}
```

Comment:
```suggestion

```
If onChange is not needed, can we consider removing it ?

Reply:
There is a warning in the console if I don't put the onChange in the Checkbox. The logic in the `onClick` can't really be used in the `onChange` because this event doesn't seem to capture keyboard event for the click+shift feature.

Reply:
I can leave a comment for that if you want

Reply:
Yes, it can be useful

Reply:
Done in the other commit